src/coroutine.rs

```diff
@@ -113,6 +122,20 @@ pub(crate) fn iter_result(result: IterNextOutput<PyObject, PyObject>) -> PyResul
 
 #[pymethods(crate = "crate")]
 impl Coroutine {
+    #[getter]
+    fn __name__(&self) -> PyResult<Py<PyString>> {
+        self.name
+            .clone()
```

Comment:
Not sure if it is worth it, but using [`clone_ref`](https://docs.rs/pyo3/latest/pyo3/prelude/struct.Py.html#method.clone_ref) would avoid the implicit check for the GIL.